### ISSUE TITLE
Include beam sync stats in metrics

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/trinity#1943]
### What is wrong?

The following would be nice to have in the metrics dashboard.

- beam sync lag
  - number of blocks
  - number of seconds
- after every request for a single urgent node:
  - RTT
  - 0 or 1 nodes actually received
- after every EVM pause for data: average RTT per node collected
- after each transaction applied: gas used since block beginning (and maybe % complete)
- periodic gauge reports, maybe every 3s:
  - urgent trie nodes retrieved per second
  - number of pending predictive trie node hashes
  - current RTT latency for queen peer
  - number of active predictive block imports (I don't think this is actually tracked yet)
  - highest block number being predictively executed (I don't think this is actually tracked yet)
  - how many pending tasks in the beam sync preview threadpool?
  - asyncio event loop delay (see log with "loop blocked or overloaded")
- at block import
    - percent time spent on block execution
    - time taken to import
    - average RTT per urgent trie node
    - Megagas/sec (*excluding* time waiting on trie nodes)
    - number of each different node type fetched on-demand (accounts, storages, contract code, storage nodes, and total trie nodes) See the "Beamed" log
    - total number of nodes and node type required for each block (see the log with "unique node hashes")

### How can it be fixed

Hook these values into metrics reporting.

Some things marked as "(I don't think this is actually tracked yet)" probably need to have some kind of tracking added.

**Comment by pipermerriam** [comment]
@carver feel free to expand this

**Comment by carver** [comment]
Added quite a few things that I look for in the logs (and a couple on my wishlist).
